File: algorithms/insert_sort.js
var insert = function(array, rightIndex, value) {
    var index;
    for(index=rightIndex; (index>=0)&&(value < array[index]); index--){
        array[index+1] = array[index];
    }
    array[index+1]=value;
};

var array = [3, 5, 7, 11, 13, 2, 9, 6];

insert(array, 4, 2);
println("Array after inserting 2:  " + array);
Program.assertEqual(array, [2, 3, 5, 7, 11, 13, 9, 6]);

insert(array, 5, 9);
println("Array after inserting 9:  " + array);
Program.assertEqual(array, [2, 3, 5, 7, 9, 11, 13, 6]);

insert(array, 6, 6);
println("Array after inserting 6:  " + array);
Program.assertEqual(array, [2, 3, 5, 6, 7, 9, 11, 13]);

File: algorithms/insert_sort.rb
def insert(array, right_index, value)
  index = right_index
  while (index >= 0) && (value < array[index])
    array[index+1] = array[index]
    index -= 1
  end
  array[index+1] = value
end

def insertion_sort(array)
  i = 1
  while (i < array.length)
    insert(array, i-1, array[i])
    i += 1
  end
  p array
end

array = [3, 5, 7, 11, 13, 2, 9, 6]
insertion_sort(array)
